### PR TITLE
fix(profil_controller#update_email): changing email from current_user.email to current_user.email destroy current user. whoops ☠️'

### PR DESCRIPTION
**Pull request by mfo**

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -244,9 +244,16 @@ en:
         one: User
         other: Users
     attributes:
+      default_attributes: &default_attributes
+        password: 'Le mot de passe'
+        requested_merge_into: "Nouvelle adresse email"
```
Comment (kemenaran) [inline config/locales/en.yml:249]:
```suggestion
        password: 'password'
        requested_merge_into: 'new email address'
```

File: config/locales/fr.yml [hunk 2]
```diff
@@ -244,6 +244,7 @@ fr:
     attributes:
       default_attributes: &default_attributes
         password: 'Le mot de passe'
+        requested_merge_into: "Nouvelle adresse email"
```
Comment (kemenaran) [inline config/locales/fr.yml:247]:
```suggestion
        requested_merge_into: 'La nouvelle adresse email'
```

File: config/locales/fr.yml [hunk 3]
```diff
@@ -273,6 +274,8 @@ fr:
               not_strong: 'n’est pas assez complexe'
             password_confirmation:
               confirmation: ': Les deux mots de passe ne correspondent pas'
+            requested_merge_into:
+              same: "ne peut être identique à l'ancienne"
```
Comment (kemenaran) [inline config/locales/fr.yml:278]:
```suggestion
              same: "ne peut être identique à l’ancienne"
```

**Review by kemenaran — COMMENTED** [review]
Propre, le validateur, c'est élégant 👌

File: spec/controllers/users/profil_controller_spec.rb [hunk 4]
```diff
@@ -48,6 +48,14 @@
   end
 
   describe 'PATCH #update_email' do
+    context 'when email is same as user' do
+      it 'fails' do
+        patch :update_email, params: { user: { email: user.email } }
+        expect(response).to have_http_status(302)
+        expect(flash[:alert]).to eq(["Nouvelle adresse email ne peut être identique à l'ancienne"])
```
Comment (mfo) [inline spec/controllers/users/profil_controller_spec.rb:55]:
```suggestion
        expect(flash[:alert]).to eq(["Nouvelle adresse email ne peut être identique à l’ancienne"])
```

File: spec/controllers/users/profil_controller_spec.rb [hunk 5]
```diff
@@ -48,6 +48,14 @@
   end
 
   describe 'PATCH #update_email' do
+    context 'when email is same as user' do
+      it 'fails' do
+        patch :update_email, params: { user: { email: user.email } }
+        expect(response).to have_http_status(302)
+        expect(flash[:alert]).to eq(["Nouvelle adresse email ne peut être identique à l’ancienne"])
```
Comment (mfo) [inline spec/controllers/users/profil_controller_spec.rb:55]:
```suggestion
        expect(flash[:alert]).to eq([""La nouvelle adresse email ne peut être identique à l’ancienne"])
```

File: app/models/user.rb [hunk 6]
```diff
@@ -223,12 +225,21 @@ def merge(old_user)
   end
 
   def ask_for_merge(requested_user)
-    update(requested_merge_into: requested_user)
-    UserMailer.ask_for_merge(self, requested_user.email).deliver_later
+    if update(requested_merge_into: requested_user)
+      UserMailer.ask_for_merge(self, requested_user.email).deliver_later
+      return true
```
Comment (tchak) [inline app/models/user.rb:230]:
```suggestion
      true
```

File: app/models/user.rb [hunk 7]
```diff
@@ -223,12 +225,21 @@ def merge(old_user)
...
+    else
+      return false
```
Comment (tchak) [inline app/models/user.rb:232]:
```suggestion
      false
```

**Review by tchak — APPROVED** [review]

**Pull request merged**